packages/cli/test/unit/lib/addons/resolve.unit.test.ts

```diff
@@ -0,0 +1,303 @@
+                expect(memoizedAddon).to.have.nested.include({name: 'myaddon-6'})
+              })
+          })
+          .then(function () {
+            const diffId = nock('https://api.heroku.com:443')
+              .post('/actions/addons/resolve', {app: 'myapp', addon: 'myaddon-7'}).reply(200, [{name: 'myaddon-7'}])
+
+            return resolveAddon(getHerokuAPI(), 'myapp', 'myaddon-7')
+              .then(function (diffIdAddon: Heroku.AddOn) {
+                expect(diffIdAddon).to.have.nested.include({name: 'myaddon-7'})
+                diffId.done()
+              })
+          })
+          .then(function () {
+            const diffApp = nock('https://api.heroku.com:443')
+              .post('/actions/addons/resolve', {app: 'fooapp', addon: 'myaddon-6'}).reply(200, [{name: 'myaddon-6'}])
+
+            return resolveAddon(getHerokuAPI(), 'fooapp', 'myaddon-6')
+              .then(function (diffAppAddon: Heroku.AddOn) {
+                expect(diffAppAddon).to.have.nested.include({name: 'myaddon-6'})
+                diffApp.done()
+              })
+          })
+          .then(function () {
+            const diffAddonService = nock('https://api.heroku.com:443')
+              .post('/actions/addons/resolve', {app: 'fooapp', addon: 'myaddon-6', addon_service: 'slowdb'}).reply(200, [{name: 'myaddon-6'}])
+
+            return resolveAddon(getHerokuAPI(), 'fooapp', 'myaddon-6', {addon_service: 'slowdb'})
+              .then(function (diffAddonServiceAddon: Heroku.AddOn) {
+                expect(diffAddonServiceAddon).to.have.nested.include({name: 'myaddon-6'})
+                diffAddonService.done()
+              })
+          })
+      })
+
+      it('does not memoize errors', () => {
+        const api = nock('https://api.heroku.com:443')
+          .post('/actions/addons/resolve', {app: 'myapp', addon: 'myaddon-8'}).reply(500, {id: 'internal server error'})
+
+        return resolveAddon(getHerokuAPI(), 'myapp', 'myaddon-8')
+          .then(() => {
+            throw new Error('unreachable')
+          })
+          .catch((error: any) => {
+            expect(error.body).to.have.nested.include({id: 'internal server error'})
+          })
+          .then(() => api.done())
+          .then(function () {
+            nock.cleanAll()
+
+            const apiRetry = nock('https://api.heroku.com:443')
+              .post('/actions/addons/resolve', {app: 'myapp', addon: 'myaddon-8'}).reply(200, [{name: 'myaddon-8'}])
+
+            return resolveAddon(getHerokuAPI(), 'myapp', 'myaddon-8')
+              .then((addon: Heroku.AddOn) => expect(addon).to.have.nested.include({name: 'myaddon-8'}))
+              .then(() => apiRetry.done())
+          })
+          .then(function () {
+            nock.cleanAll()
+
+            return resolveAddon(getHerokuAPI(), 'myapp', 'myaddon-8')
+              .then((addon: Heroku.AddOn) => expect(addon).to.have.nested.include({name: 'myaddon-8'}))
+          })
+      })
+    })
+  })
```

Comment:
This test initially used a 403 and `two_factor' error, which is now automatically retried within @heroku-cli/command's APIClient. This caused the test to hang on a ux.prompt() call and would require refactoring the test. I believe this 500 error fits the initial intent, but I'm not 100% certain